cmake/Modules/CompilerConfig.cmake

```diff
@@ -84,16 +84,27 @@ else()
   endif()
 
   add_compile_options(
+    -Werror
```

Comment:
`-Werror` is a setting for CI where known compilers are used. Enabling it in user space just make just the source code incompatible with future or unknown compilers (very annoying bisecting code and returning to old revision with already fixed warnings) and you always have to disable the flag when experimenting (unfinished features will likely generate warnings). Any reasons why not just enable this on CI to forbid code that generates warning to be submitted? I get the point that some annoyed users might be more likely to submit PRs for a warning when they use a untested compiler, but wouldn't it still be better to just try to cover most compilers in CI?

Reply:
We could add an cmake option to optionally disable it locally, although after using it recently, I've found that I've managed to get used to it while programming something. It's mostly only been a very minor annoyance if anything, and we disable the warnings that we don't feel make sense for us. I know that future compiler versions may be unpredictable with it, but I generally don't think it's a big deal either. But this is just my opinion on the matter.

Reply:
I would appreciate a option to disable them, but temporarily disabling them locally wouldn't be a big deal for me. I normally use nightly compilers and prefer to have the warnings on a maximum which I might ignore for a test run but when making a commit.